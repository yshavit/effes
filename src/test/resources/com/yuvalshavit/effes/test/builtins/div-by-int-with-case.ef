type DivByZero

divide10 (denominator: Int) -> Int | DivByZero:
  return case denominator of
    IntZero: DivByZero
    IntValue: 10 . / denominator

main:
  print divide10 2
  print divide10 0
