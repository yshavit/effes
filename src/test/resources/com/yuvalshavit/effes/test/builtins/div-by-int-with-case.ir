def divide10 (denominator: IntValue | IntZero) -> DivByZero | IntValue | IntZero:
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: denominator
          pos: 0
      of IntZero:
        CtorInvoke
          type: DivByZero
      of IntValue (exhaustive):
        MethodInvoke
          method: IntValue::/ [built-in]
          arg0:
            IntLiteral
              10
          arg1:
            VarExpression
              arg: denominator
              pos: 0
