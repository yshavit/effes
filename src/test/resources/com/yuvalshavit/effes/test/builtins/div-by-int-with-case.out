Your argument was: 5
Your argument was: DivByZero
>> null
